src/mlpack/core/optimizers/fw/frank_wolfe_impl.hpp

```diff
@@ -42,6 +42,11 @@ template<typename FunctionType>
 double FrankWolfe<LinearConstrSolverType, UpdateRuleType>::
 Optimize(FunctionType& function, arma::mat& iterate)
 {
+  static_assert(mlpack::static_checks::CheckEvaluate<FunctionType>::value,
+      "The FunctionType does not have a correct definition of Evaluate.");
```

Comment:
I'm wondering if it makes sense to improve the error message, and give some additional hints e.g. about the correct function definition. This is somehow related to: https://github.com/mlpack/mlpack/issues/891, let me know what you think, also we don't have to do it now.

Reply:
Sure, I'll add the correct definition and maybe a reference to the `FunctionType` API documentation(?)
The documentation needs some updating, I guess I'll do that in a separate PR. :)

Reply:
This is a minor comment, but I do think maybe we should use `Evaluate()` instead of `Evaluate` throughout to make it clear that it is a method that's required.  If we want to make the error messages more complex, maybe it is worthwhile to add some method to `CheckEvaluate<>` so that the error message is consistent across checks.